proofs.go

```diff
@@ -166,7 +171,12 @@ type ProofD struct {
 
 // MergeProofP merges a ProofP into the ProofD.
 func (p *ProofD) MergeProofP(proofP *ProofP, _ *gabikeys.PublicKey) {
-	p.SecretKeyResponse().Add(p.SecretKeyResponse(), proofP.SResponse)
+	if proofP.P == nil { // new protocol version
```

Comment:
I agree that this approach for dealing with different protocol versions doesn't scale, but if we want to have access to that version number in this particular method, that would mean that the version number would have to be either a new field on `ProofD`, or a parameter to `MergeProofP`. Neither would be pretty.

Additionally, of course I might be completely wrong but I don't reallly expect further versions of the protocol at all. The old one was the very first one, since 2016 or so.

For now I would prefer to keep this the way it is, and instead deal with version numbers if and when yet another protocol version would become necessary.